routers/user/getOneStudent.ts

```diff
@@ -0,0 +1,45 @@
+import { NextFunction, Request, Response } from "express";
+import { Roles } from "../../src/entities/types/Roles";
+import { ValidationError } from "../../utils/errorsHandler";
+import { StudentsRating } from "../../src/entities/studentsRating/studentsRating.entity";
+import { myDataSource } from "../../config/database.configuration";
+import { StudentStatus } from "../../src/entities/types/studentsData";
+import { UserPayloadData } from "../../utils/createTokens";
+import { DestructuringToGetOne } from "../../utils/destructuringGetOne";
+
+type RequestAndPayloadUser = Request & UserPayloadData;
+
+export const getOneStudent = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { id, role } = req.user as RequestAndPayloadUser;
+
+    if (role !== Roles.HR && role !== Roles.STUDENT) throw new ValidationError('Access denied.', 401);
+
+    const userId = req.params.id;
+
+    const results = await myDataSource
+      .getRepository(StudentsRating)
+      .createQueryBuilder('studentsRating')
+      .leftJoinAndSelect('studentsRating.user', 'user')
+      .leftJoinAndSelect('studentsRating.studentsData', 'studentsData')
+      .where(`user.role = '${Roles.STUDENT}'`)
+      .andWhere(`studentsData.status = '${StudentStatus.AVAILABLE}'`)
+      .andWhere('studentsRating.id = :userId', { userId })
+      .orderBy('studentsData.lastName, studentsData.firstName')
```

Comment:
dose we need it? You take only one student

Reply:
nie rozumiem. piszesz o getOne() ?